gems/pending/disk/modules/miq_disk_cache.rb

```diff
@@ -0,0 +1,109 @@
+require 'rufus/lru'
+require_relative "../MiqDisk"
+require 'ostruct'
+
+module MiqDiskCache
+  MIN_SECTORS_TO_CACHE = 32
+  DEF_CACHE_SIZE       = 100
+  DEBUG_CACHE_STATS    = false
+
+  attr_reader :d_size, :blockSize, :cache_size, :sectors_to_cache, :downstream
+
+  def self.new(downstream, d_info = nil)
+    raise "MiqDiskCache: Downstream Disk Module is nil" if downstream.nil?
+    @dInfo                  = d_info || OpenStruct.new
+    @dInfo.cache_size       = @dInfo.cache_size || DEF_CACHE_SIZE
+    @dInfo.sectors_to_cache = @dInfo.sectors_to_cache || MIN_SECTORS_TO_CACHE
+    @dInfo.block_size       = downstream.blockSize
+    @dInfo.down_stream      = downstream
+
+    MiqDisk.new(self, @dInfo, 0)
+  end
+
+  def d_init
+    @block_cache      = LruHash.new(@dInfo.cache_size)
+    @cache_hits       = Hash.new(0)
+    @cache_misses     = Hash.new(0)
+    @blockSize        = @dInfo.block_size
```

Comment:
Same here...as this is a new class, let's prefer `@block_size` as the variable name.